ui/components/app/simulation-preview/simulation-preview.tsx

```diff
@@ -0,0 +1,190 @@
+  if (!simulationData) {
+    return [];
+  }
+  const { nativeBalanceChange } = simulationData;
+  const balanceChanges = [];
+
+  if (nativeBalanceChange) {
+    balanceChanges.push(getNativeAssetBalanceChange(nativeBalanceChange));
+  }
+  return balanceChanges;
+}
+
+/**
+ * Content when simulation has failed.
+ */
+const SimulationFailedContent: React.FC = () => {
+  const t = useI18nContext();
+  return (
+    <Text
+      color={TextColor.warningDefault}
+      variant={TextVariant.bodySm}
+      display={Display.Flex}
+      alignItems={AlignItems.flexStart}
+    >
+      <Icon name={IconName.Warning} marginInlineEnd={1} />
+      {t('simulationPreviewFailed')}
+    </Text>
+  );
+};
+
+/**
+ * Content when there are no balance changes.
+ */
+const NoBalanceChangesContent: React.FC = () => {
+  const t = useI18nContext();
+  return (
+    <Text
+      color={TextColor.textAlternative}
+      variant={TextVariant.bodySm}
+      display={Display.Flex}
+      alignItems={AlignItems.flexStart}
+    >
+      {t('simulationPreviewNoBalanceChanges')}
+    </Text>
+  );
+};
+
+/**
+ * Header at the top of the simulation preview.
+ */
+const SimulationPreviewHeader: React.FC = () => {
+  const t = useI18nContext();
+  return (
+    <Box
+      display={Display.Flex}
+      flexDirection={FlexDirection.Row}
+      alignItems={AlignItems.center}
+      gap={1}
+      marginBottom={3}
+    >
+      <Text variant={TextVariant.bodyMdMedium}>
+        {t('simulationPreviewTitle')}
+      </Text>
+      <InfoTooltip
+        position="right"
+        iconFillColor="var(--color-icon-muted)"
+        contentText={t('simulationPreviewTitleTooltip')}
+      />
+    </Box>
+  );
+};
+
+/**
+ * Layout component for the simulation preview.
+ *
+ * @param options0
+ * @param options0.children
+ */
+const SimulationPreviewLayout: React.FC = ({ children }) => (
+  <Box
+    display={Display.Flex}
+    flexDirection={FlexDirection.Column}
+    borderRadius={BorderRadius.MD}
+    borderColor={BorderColor.borderDefault}
+    padding={3}
+    margin={4}
+  >
+    {children}
+  </Box>
+);
+
+/**
+ * Preview of a transaction's effects using simulation data.
+ *
+ * @param props
+ * @param props.simulationData - The simulation data to display.
+ */
+export const SimulationPreview: React.FC<SimulationPreviewProps> = ({
+  simulationData,
+}: SimulationPreviewProps) => {
+  const t = useI18nContext();
+
+  const simulationFailed = !simulationData;
+  if (simulationFailed) {
+    return (
+      <SimulationPreviewLayout>
+        <SimulationPreviewHeader />
+        <SimulationFailedContent />
+      </SimulationPreviewLayout>
+    );
+  }
+
+  const balanceChanges = getAssetBalanceChanges(simulationData);
+  if (balanceChanges.length === 0) {
+    return (
+      <SimulationPreviewLayout>
+        <SimulationPreviewHeader />
+        <NoBalanceChangesContent />
+      </SimulationPreviewLayout>
+    );
+  }
```

Comment:
Nit:

Can we do some refactor here? as 
```
 <SimulationPreviewLayout>
      <SimulationPreviewHeader />
 ```
Are the same across the conditions, also `<NoBalanceChangesContent />` and `<SimulationFailedContent />` have great similarities ... the `Text` wrapper are the same as 
```
<Text
      color={TextColor.textAlternative}
      variant={TextVariant.bodySm}
      display={Display.Flex}
      alignItems={AlignItems.flexStart}
    >
```

Reply:
Thank you, Danica. Good observation on the repetition! If it is okay with you, I will address your comment in [an already existing follow-up PR](https://github.com/MetaMask/metamask-extension/pull/23529) as the code has [changed significantly](https://github.com/MetaMask/metamask-extension/pull/23529/files#diff-e3a4ad038cfdd55d3f45e91642d0a5816b2aa1a54a2fc49dd31e1d65b1c26a9b).